xml/docker_concept.xml

```diff
@@ -0,0 +1,498 @@
+      </remark>
+      A <emphasis>Dockerfile</emphasis> stores changes made on top of the base
+      image. The <xref linkend="Docker_engine"/> reads instructions in the
+      <emphasis>Dockerfile</emphasis> and builds a new image according to the
+      instructions.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Container</term>
+    <listitem>
+     <remark>toms 2019-02-04: according to
+      https://docs.docker.com/glossary/?term=container a container consists
+      of a) a Docker image, b) an executation environment, and c) a set of
+      instructions. Maybe add that info after your first sentence?
+     </remark>
+     <para>
+      A <emphasis>container</emphasis> is a running<remark>toms 2019-02-04:
+       and a idle/stopped image is not a container? ;) Isn't that term a bit
+       too restrictive?</remark> instance based on a
+      particular &docker; <xref linkend="Docker_Image"/>. Each
+      <emphasis>container</emphasis> can be distinguished by a unique container
+      ID.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Registry</term>
+    <listitem>
+     <para>
+      A <emphasis>registry</emphasis> is storage for already created images. It
+      typically contains several <emphasis>repositories</emphasis> There are two
+      types of registry:
+     </para>
+     <itemizedlist>
+      <listitem>
+       <formalpara>
+        <title>Public Registry:</title>
+        <para>
+        where everyone (usually registered) can download and
+        use images. A typical public registry is
+        <link xlink:href="https://hub.docker.com/">Docker Hub</link>.
+        </para>
+       </formalpara>
+      </listitem>
+      <listitem>
+       <formalpara>
+        <title>Private Registry:</title>
+        <para>
+        these are accessible for particular users or from a
+        particular private network.
+        </para>
+       </formalpara>
+      </listitem>
+     </itemizedlist>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term>Repository</term>
+    <listitem>
+     <para>
+      A <emphasis>repository</emphasis> is storage in a
+      <emphasis>registry</emphasis> that stores a different version of a
+      particular image. You can pull or push images from or to a repository.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry xml:id="vle.docker.cgroup">
+    <term>Control group</term>
+    <listitem>
+     <para>
+      <emphasis>Control groups</emphasis>, also called
+      <literal>cgroups</literal>, is a Linux kernel feature that allows
+      aggregating or partitioning tasks (processes) and all their children into
+      hierarchically organized groups to isolate resources.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry xml:id="vle.docker.namespace">
+    <term>Namespace</term>
+    <listitem>
+     <para>
+      &deng; uses <emphasis>namespaces</emphasis> for its containers that
```

Comment:
This sentence is a bit ambiguous (it's not clear to what 'that' refers: who isolates the resources?) Also: what are resources isolated *from*? from other resources reserved for other containers? 
To make it clearer, you could put it like that: "Used by &deng; for its containers to isolate resources reserved for particular containers (from???)."

Reply:
Fixed.
For more information about Docker namespaces, see [this article](https://success.docker.com/article/introduction-to-user-namespaces-in-docker-engine).

Reply:
They are isolated from other containers or the host system.
https://success.docker.com/article/introduction-to-user-namespaces-in-docker-engine